build/Rulesets/Roslyn_BuildRules.ruleset

```diff
@@ -77,6 +82,7 @@
     <Rule Id="RS1012" Action="None" />
     <Rule Id="RS1013" Action="None" />
     <Rule Id="RS1014" Action="Warning" />
+    <Rule Id="RS1022" Action="None" /> <!-- https://github.com/dotnet/roslyn/issues/26400: Enable rules CA1062, CA1303, CA1508, CA2100 and CA2215 for Roslyn.sln -->
```

Comment:
@jcouv `RS1022` is the new analyzer rule that flags analyzers that access any APIs from Workspace layer (DoNotUseTypesFromAssemblyRuleId). Given that lot of our IDE analyzers use backdoor access to Workspace options, we cannot turn it on until @sharwell is done moving all IDE analyzers to NuGet. Can you please file a separate issue to track enabling this rule and assign it to Sam?

Reply:
Filed https://github.com/dotnet/roslyn/issues/26420 and updated comment accordingly